sound/soc/sof/ipc4-topology.h

```diff
@@ -473,6 +473,7 @@ struct sof_ipc4_process {
 	struct sof_ipc4_msg msg;
 	u32 base_config_ext_size;
 	u32 init_config;
+	u32 buffer_period_size;
```

Comment:
commit message: you're describing a solution without explaining what the issue is. Why do we need this in the first place? In what context does this help?

Reply:
This is intended to handle DP module that requires larger buffer than 1ms.

Reply:
DP modules scheduling is based on their IBS and OBS

In general:

1) the module is ready to run and will be scheduled if it has at least IBS number of bytes available on each input pin and at least OBS number of bytes free on each output pin
please note that each pin may have its own format - number of channels, sampling frequency, etc. so each pin may have its own IBS/OBS

IBS and OBS are as declared in module bind IPC. 

2) calculation of module period - required for synchronization with LL modules and EDF scheduling - is also based on IBS(IBSes)

special case like variable bitrate: the module may override both 1 and 2 if really needed. 

so - for DP modules IBS/OBS values should be precise, it is critical for proper data flow. 

example: if a module has 2 input pins, one 16000/16/2, second 48000/16/2 and one output pin 48000/32/4 and is desired to be scheduled every 10ms: 

in pin1 IBS should be 16(samples/1ms) * 2 (bytes per sample) * 2 (channels) * 10(ms) = 640
in pin2 IBS should be 48(samples/1ms) * 2 (bytes per sample) * 2 (channels) * 10(ms) = 1920
out pin1 OBS should be 48(samples/1ms) * 4 (bytes per sample) * 4 (channels) * 10(ms) = 7680

another note - bigger OBS won't hurt the processing but is a memory waste - so should be avoided.